bentoml/yatai/python_api.py

```diff
@@ -149,3 +164,193 @@ def update_bento_upload_progress(
+        return ApplyDeploymentResponse(status=Status.ABORTED(str(error)))
+    except BentoMLException as error:
+        return ApplyDeploymentResponse(status=Status.INTERNAL(str(error)))
+
+
+# TODO update_deployment is not finished.  It will be working on along with cli command
+def update_deployment(deployment_name, namespace, yatai_service=None):
+    raise NotImplemented
+
+
+def apply_deployment(deployment_info, yatai_service=None):
+    if yatai_service is None:
+        from bentoml.yatai import get_yatai_service
+
+        yatai_service = get_yatai_service()
+
+    try:
+        if isinstance(deployment_info, dict):
+            deployment_pb = deployment_dict_to_pb(deployment_info)
+        elif isinstance(deployment_info, str):
+            deployment_pb = deployment_yaml_string_to_pb(deployment_info)
+        else:
+            raise BentoMLDeploymentException(
+                'Currently, we do not handle deployment info type {}'.format(
+                    str(type(deployment_info))
+                )
+            )
+
+        validation_errors = validate_deployment_pb_schema(deployment_pb)
+        if validation_errors:
+            return ApplyDeploymentResponse(
+                status=Status.ABORTED(
+                    'Failed to validate deployment. {errors}'.format(
+                        errors=validation_errors
+                    )
+                )
+            )
+
+        return yatai_service.ApplyDeployment(
+            ApplyDeploymentRequest(deployment=deployment_pb)
+        )
+    except BentoMLDeploymentException as error:
+        return ApplyDeploymentResponse(status=Status.ABORTED(str(error)))
+    except BentoMLException as error:
+        return ApplyDeploymentResponse(status=Status.INTERNAL(str(error)))
+
+
+def describe_deployment(namespace, name, yatai_service=None):
+    if yatai_service is None:
+        from bentoml.yatai import get_yatai_service
+
+        yatai_service = get_yatai_service()
+    return yatai_service.DescribeDeployment(
+        DescribeDeploymentRequest(deployment_name=name, namespace=namespace)
+    )
+
+
+def get_deployment(namespace, name, yatai_service=None):
+    if yatai_service is None:
+        from bentoml.yatai import get_yatai_service
+
+        yatai_service = get_yatai_service()
+    return yatai_service.GetDeployment(
+        GetDeploymentRequest(deployment_name=name, namespace=namespace)
+    )
+
+
+def delete_deployment(
+    deployment_name, namespace, force_delete=False, yatai_service=None
+):
+    if yatai_service is None:
+        from bentoml.yatai import get_yatai_service
+
+        yatai_service = get_yatai_service()
+
+    return yatai_service.DeleteDeployment(
+        DeleteDeploymentRequest(
+            deployment_name=deployment_name,
+            namespace=namespace,
+            force_delete=force_delete,
+        )
+    )
+
+
+def list_deployments(
+    limit, filters, labels, namespace, is_all_namespaces, yatai_service=None
```

Comment:
instead of ```namespace, is_all_namespaces```, let's make it ```namespace=ALL_NAMESPACE_TAG```, which means if user don't specify a namespace, it's going to list for all namespaces.

Reply:
similarily, make limit, filter, labels optional parameters -  Note that all these are suppose to be user facing APIs, let's make it more concise and easier to understand